go/libraries/doltcore/sqle/dolt_session.go

```diff
@@ -695,6 +716,21 @@ func (sess *DoltSession) AddDB(ctx *sql.Context, db sql.Database, dbData env.DbD
 	return nil
 }
 
+// CreateTemporaryTablesRoot creates an empty root value and a table edit session for the purposes of storing
+// temporary tables. This should only be used on demand. That is only when a temporary table is created should we
+// create the root map and edit session map.
+func (sess *DoltSession) CreateTemporaryTablesRoot(ctx *sql.Context, dbName string, ddb *doltdb.DoltDB) error {
+	newRoot, err := doltdb.EmptyRootValue(ctx, ddb.ValueReadWriter())
+	if err != nil {
+		return err
+	}
+
+	sess.tempTableRoots[dbName] = newRoot
```

Comment:
Should call session.SetTempTableRoot here instead of direct asssignment (after you create the edit session below)